README.md

```diff
@@ -134,84 +129,123 @@ The priority of config values is as follows:
 
 **Note:** we are using [`@operational/components`](https://github.com/contiamo/operational-components) for this example, but a UI of config values _can_ be assembled with any UI library, or even with plain ole HTML-tag JSX.
 
 ```ts
 // pages/ConfigurationPage.tsx
 import { Page, Card, Input, Button, Checkbox } from "@operational/components";
-import { AdminConfig } from "./components/Config";
-
-export default () => (
-  <Page title="Configuration">
-    <Card title="Configuration">
-      <AdminConfig>
-        {({ fields, onFieldChange, submit, reset }) => (
-          <>
-            {fields.map(field =>
-              field.type === "boolean" ? (
-                <Checkbox
-                  key={field.path}
-                  value={field.value}
-                  label={field.path}
-                  onChange={val => onFieldChange(field.path, val)}
-                />
-              ) : (
-                <Input
-                  key={field.path}
-                  value={field.value}
-                  label={field.path}
-                  onChange={val => onFieldChange(field.path, val)}
-                />
-              ),
-            )}
-            <Button onClick={submit}>Update config</Button>
-            <Button onClick={reset}>Reset config</Button>
-          </>
+import { useAdminConfig } from "./components/Config";
+
+export default () => {
+  const { fields, reset } = useAdminConfig();
+
+  return (
+    <Page title="Configuration">
+      <Card title="Configuration">
+        {fields.map(field =>
+          field.type === "boolean" ? (
+            <Checkbox key={field.key} value={field.value} label={field.key} onChange={field.set} />
+          ) : (
+            <Input key={field.key} value={field.value} label={field.key} onChange={field.set} />
+          ),
         )}
-      </AdminConfig>
-    </Card>
-  </Page>
-);
+        <Button onClick={reset}>Reset config</Button>
+      </Card>
+    </Page>
+  );
+};
 ```
 
 You have also access to `field.windowValue` and `field.storageValue` if you want implement more advanced UX on this page.
 
+## Multiconfiguration admin page
+
+As soon as you have more than one configuration in your project, creating an administration page than merge everything is a nice to have. Of course, you will want a kind of `ConfigSection` component that take the result of any `useAdminConfig()` (so `field`, `reset` and `namespace` as props).
+
+Spoiler alert, having this kind of component type safe can be tricky, indeed you can try use `ReturnType<typeof useFirstAdminConfig> | ReturnType<typeof useSecondAdminConfig>` as props but typescript will fight you (`Array.map` will tell you that the signature are not compatible).
+
+Anyway, long story short, this library provide you an easy way to with this: `GenericAdminFields` type. This type is compatible with every configuration and will provide you a nice framework to create an amazing UX.
+
```

Comment:
Since it is a public lib could be nice to have a native speaker proof-read the doc update :)